src/com/softwaret/kpdf/util/extension/ApplicationCallExtension.kt

```diff
@@ -14,3 +19,28 @@ val ApplicationCall.userLoginFromPrincipal
 
 suspend fun ApplicationCall.respondWith(response: Response) =
     respond(status = response.code, message = response.body)
+
+internal class Form(private val values: HashMap<String, String>) {
+
+    inline fun <reified T> get(create: (String) -> T) =
+        (values[T::class.simpleName?.toLowerCase()] ?: "").let(create)
+}
+
+internal suspend fun ApplicationCall.receiveForm(context: CoroutineContext, data: (Form) -> Response) =
+    withContext(context) {
+        val values = hashMapOf<String, String>()
+        receiveMultipart().forEachPart { part ->
+            when (part) {
+                is PartData.FileItem -> {
```

Comment:
Mam troszkę vibe że pobieranie pliku po to żeby go zencodować do Base64 i potem zdecodować jest trochę niewydajne? Nie jestem pewien ale przydałoby się sprawdzić jak to robią ludzie na legitnych backendach

Reply:
True, zorientuję się i dam znać